services/blockstorage/service.go

```diff
@@ -297,10 +333,55 @@ func (s *service) validateProtocolVersion(blockPair *protocol.BlockPairContainer
 }
 
 // TODO: this should not be called directly from CommitBlock, it should be called from a long living goroutine that continuously syncs the state storage
-func (s *service) syncBlockToStateStorage(commitedBlockPair *protocol.BlockPairContainer) error {
+func (s *service) syncBlockToStateStorage(committedBlockPair *protocol.BlockPairContainer) error {
 	_, err := s.stateStorage.CommitStateDiff(&services.CommitStateDiffInput{
-		ResultsBlockHeader: commitedBlockPair.ResultsBlock.Header,
-		ContractStateDiffs: commitedBlockPair.ResultsBlock.ContractStateDiffs,
+		ResultsBlockHeader: committedBlockPair.ResultsBlock.Header,
+		ContractStateDiffs: committedBlockPair.ResultsBlock.ContractStateDiffs,
 	})
 	return err
 }
+
+func (s *service) validateWithConsensusAlgos(prevBlockPair *protocol.BlockPairContainer, lastCommittedBlockPair *protocol.BlockPairContainer) error {
+	for _, handler := range s.consensusBlocksHandlers {
+		_, err := handler.HandleBlockConsensus(&handlers.HandleBlockConsensusInput{
```

Comment:
Once my PR gets merged: https://github.com/orbs-network/orbs-network-go/pull/154
This line needs to be fixed to add the `Mode` argument. Open an issue to remember to do this or merge the other PR first into your PR.

The mode needs to be `VERIFY_AND_UPDATE`